tmva/tmva/src/Tools.cxx

```diff
@@ -270,52 +271,26 @@ void TMVA::Tools::ComputeStat( const std::vector<TMVA::Event*>& events, std::vec
 
 TMatrixD* TMVA::Tools::GetSQRootMatrix( TMatrixDSym* symMat )
 {
-   Int_t n = symMat->GetNrows();
-
-   // compute eigenvectors
-   TMatrixDSymEigen* eigen = new TMatrixDSymEigen( *symMat );
-
-   // D = ST C S
-   TMatrixD* si = new TMatrixD( eigen->GetEigenVectors() );
-   TMatrixD* s  = new TMatrixD( *si ); // copy
-   si->Transpose( *si ); // invert (= transpose)
-
-   // diagonal matrices
-   TMatrixD* d = new TMatrixD( n, n);
-   d->Mult( (*si), (*symMat) ); (*d) *= (*s);
-
-   // sanity check: matrix must be diagonal and positive definit
-   Int_t i, j;
-   Double_t epsilon = 1.0e-8;
-   for (i=0; i<n; i++) {
-      for (j=0; j<n; j++) {
-         if ((i != j && TMath::Abs((*d)(i,j))/TMath::Sqrt((*d)(i,i)*(*d)(j,j)) > epsilon) ||
-             (i == j && (*d)(i,i) < 0)) {
-            //d->Print();
-            Log() << kWARNING << "<GetSQRootMatrix> error in matrix diagonalization; printed S and B" << Endl;
-         }
-      }
+   TDecompChol chol(*symMat);
+   if (!chol.Decompose())
+   {
+      Log() << kWARNING << "<GetSQRootMatrix> matrix not positive definite; printed S and B" << Endl;
```

Comment:
What does it mean "printed S and B" ?

Reply:
Who knows, it's part of the original code. Please propose what to do.